reflect_struct_decoder.go

```diff
@@ -1042,7 +1042,7 @@ func (decoder *stringModeNumberDecoder) Decode(ptr unsafe.Pointer, iter *Iterato
 	}
 	c = iter.readByte()
 	if c != '"' {
-		iter.ReportError("stringModeNumberDecoder", `expect ", but found `+string([]byte{c}))
+		iter.ReportError("stringModeNonStringDecoder", `expect ", but found `+string([]byte{c}))
```

Comment:
Maybe it needs a testcase to cover these new added lines to avoid the coverage diff reported by codecov.